src/items/item.h

```diff
@@ -1036,8 +1039,43 @@ class Item : virtual public Thing
 			return !parent || parent->isRemoved();
 		}
 
-		uint32_t getImbuement(uint8_t slot);
-		void setImbuement(uint8_t slot, int64_t info);
+		/**
+		 * @brief Get the Imbuement Info object
+		 *
+		 * @param slot
+		 * @param imbuementInfo (Imbuement *imbuement, uint32_t duration = 0)
+		 * @return true = duration is > 0 (info >> 8)
+		 * @return false
+		 */
+		bool getImbuementInfo(uint8_t slot, ImbuementInfo *imbuementInfo);
+		void setImbuement(uint8_t slot, uint16_t id, int32_t duration);
+		bool hasImbuementType(ImbuementTypes_t imbuementType, uint16_t maxTier) {
+			auto it = items[id].imbuementTypes.find(imbuementType);
+			if (it != items[id].imbuementTypes.end()) {
+				return (it->second >= maxTier);
+			}
+			return false;
+		}
+		bool hasImbuementCategoryId(uint16_t categoryId);
+		bool hasImbuements() {
+			for (uint8_t slotid = 0; slotid < getImbuementSlot(); slotid++) {
+				ImbuementInfo imbuementInfo;
+				if (getImbuementInfo(slotid, &imbuementInfo)) {
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		// Check if item have critical chance skill
+		bool hasCriticalChanceSkill(int32_t skill) {
```

Comment:
This function makes no sense, you can just use skill == SKILL_CRITICAL_HIT_CHANCE directly. This is already a logical expression and returns bool. No need of a function